tests/processor/test_main.py

```diff
@@ -0,0 +1,442 @@
+from assayist.common.models.source import SourceLocation
+from assayist.common.models.content import Artifact, Build
+
+# RPM INFO BLOBS
+VIM_1_2_3 = {'buildroot_id': '1',
+             'id': '1',
+             'name': 'vim',
+             'epoch': '1',
+             'version': '2',
+             'release': '3.el7',
+             'arch': 'x86_64',
+             'payloadhash': '1234567890'}
+
+VIM_2_3 = {'buildroot_id': '1',
+           'id': '2',
+           'name': 'vim',
+           'epoch': None,
+           'version': '2',
+           'release': '3.el7',
+           'arch': 'x86_64',
+           'payloadhash': '1234567890'}
+
+SSH_9_8_7 = {'buildroot_id': '2',
+             'id': '3',
+             'name': 'ssh',
+             'epoch': '9',
+             'version': '8',
+             'release': '7.el7',
+             'arch': 'x86_64',
+             'payloadhash': '0987654321'}
+
+NETWORKMANAGER_5_6_7_X86 = {'id': '4',
+                            'name': 'NetworkManager',
+                            'epoch': '5',
+                            'version': '6',
+                            'release': '7.el7',
+                            'arch': 'x86_64',
+                            'payloadhash': '112346754'}
+
+NETWORKMANAGER_5_6_7_PPC = {'id': '5',
+                            'name': 'NetworkManager',
+                            'epoch': '5',
+                            'version': '6',
+                            'release': '7.el7',
+                            'arch': 'ppc64le',
+                            'payloadhash': '752245171'}
+
+GCC_2_3_4 = {'id': '6',
+             'name': 'gcc',
+             'epoch': '2',
+             'version': '3',
+             'release': '4.el7',
+             'arch': 'x86_64',
+             'payloadhash': '1234567890'}
+
+PYTHON_3_6_7 = {'id': '7',
+                'name': 'python',
+                'epoch': '3',
+                'version': '6',
+                'release': '7.el7',
+                'arch': 'x86_64',
+                'payloadhash': '1234567890'}
+
+# ARCHIVE BLOBS
+IMAGE1 = {'id': '1',
+          'checksum': '1234567890',
+          'filename': 'image1.tar.gz',
+          'buildroot_id': '1',
+          'extra': {'image': {'arch': 'x86_64'}}}
+
+IMAGE2 = {'id': '2',
+          'checksum': '0987654321',
+          'filename': 'image2.tar.gz',
+          'buildroot_id': '2',
+          'extra': {'image': {'arch': 'ppc64le'}}}
+
+
+SOURCE_URL = "git://pkgs.devel.redhat.com/containers/virt-api#e9614e8eed02befd8ed021fe9591f8453422"
+
+
+# Test the basic function of the get_or_create methods
+def test_get_or_create_build():
+    """Test the basic function of get_or_create_build."""
+    analyzer = main_analyzer.MainAnalyzer()
+    build = analyzer.get_or_create_build('1', 'type1')
+    assert build.id_ == '1'
+    assert build.type == 'type1'
+    build.id  # exists, meaning it's been saved
+
+    build2 = analyzer.get_or_create_build('1', 'type2')
```

Comment:
This test also shows that we don't validate the build type for an existing build. Should we? (I'm fine with "no", just want to ask)

Reply:
I am unopinionated here. Realistically though we're mostly just writing through strings that we got from Brew; validation would only really matter if there was something sensible we could do if we detect a weird case. If a build is not one of the expected types what would we do differently?

Reply:
That's fine, just wanted to make sure this wasn't a mistake.